pkg/quota-controller/profile/profile_controller.go

```diff
@@ -0,0 +1,175 @@
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	schedv1alpha1 "sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/apis/quota/v1alpha1"
+	utilclient "github.com/koordinator-sh/koordinator/pkg/util/client"
+)
+
+const Name = "quotaprofile"
+
+// QuotaProfileReconciler reconciles a QuotaProfile object
+type QuotaProfileReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+func (r *QuotaProfileReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// reconcile for 2 things:
+	//   1. ensuring the Quota exists if the QuotaProfile exists
+	//   2. update Quota Spec
+	_ = log.FromContext(ctx, "quota-profile-reconciler", req.NamespacedName)
+
+	// get the profile
+	profile := &v1alpha1.ElasticQuotaProfile{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, profile)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			klog.Errorf("failed to find profile %v, error: %v", req.NamespacedName, err)
+			return ctrl.Result{Requeue: true}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(profile.Spec.NodeSelector)
+	if err != nil {
+		klog.Errorf("failed to convert profile %v nodeSelector, error: %v", req.NamespacedName, err)
+		return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, err
+	}
+	nodeList := &corev1.NodeList{}
+	if err := r.Client.List(context.TODO(), nodeList, &client.ListOptions{LabelSelector: selector}, utilclient.DisableDeepCopy); err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	// TODO: consider node status.
+	totalResource := corev1.ResourceList{}
+	for _, node := range nodeList.Items {
+		totalResource = quotav1.Add(totalResource, node.Status.Allocatable)
+	}
+
+	quotaExist := true
+	quota := &schedv1alpha1.ElasticQuota{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: profile.Namespace, Name: profile.Spec.QuotaName}, quota)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			quota = &schedv1alpha1.ElasticQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: profile.Namespace,
+					Name:      profile.Spec.QuotaName,
+				},
+			}
+			quotaExist = false
+		} else {
+			return ctrl.Result{Requeue: true, RequeueAfter: 1 * time.Second}, err
+		}
+	}
+	oldQuota := quota.DeepCopy()
+
+	resourceKeys := []string{"cpu", "memory"}
+	raw, ok := profile.Annotations[extension.AnnotationResourceKeys]
+	if ok {
+		err := json.Unmarshal([]byte(raw), &resourceKeys)
+		if err != nil {
+			klog.Warningf("failed unmarshal quota.scheduling.koordinator.sh/resource-keys %v", raw)
+		}
+	}
+
+	min := corev1.ResourceList{}
+	for _, key := range resourceKeys {
+		resourceName := corev1.ResourceName(key)
+		quantity, ok := totalResource[resourceName]
+		if ok {
+			min[resourceName] = quantity
+		}
+	}
+
+	quota.Spec.Min = min
```

Comment:
From Sum(node Allocatable) ---> min , we need a fragmentation factor here, and this factor is configurable.